gusto/state.py

```diff
@@ -253,9 +258,12 @@ def dump(self, t=0, pickup=False):
                 self.dumpfile_ll.write(*self.to_dump_latlon)
 
             # compute diagnostics
+            diagnostic_fns = ['min', 'max', 'rms', 'l2']
```

Comment:
This feels like the wrong place to enumerate these names.  Basically you're saying what diagnostic quantities the `Diagnostics` class can compute.  I think it would be better to add a field to the `Diagnostics` class that says what these are.

```python
class Diagnostics

    available_diagnostics = ["min", "max", "rms", "l2"]
    ...
```

And then here we do:
```python
for fn in self.diagnostics.available_diagnostics:
   ...
```

That way as well, you could later extend `State` to say which diagnostics you want to compute.